docs/developers/CwlParsing.md

```diff
@@ -0,0 +1,40 @@
+# How CWL is Parsed
+
+The process of parsing CWL runs approximately like this:
+
+1. Produce a canonical one-file representation of the CWL as a JSON object in memory 
+1. Use Circe to turn that representation into scala case classes
+1. Convert the case classes into an appropriate WOM representations
+
+## Getting a canonical flat file
+
+### From a source file and dependencies zip
```

Comment:
Just to note that the process is the same when running Cromwell in `run` mode. Even if no dependency zip is provided, it will still perform the same recursive flattening starting from root cwl file passed on the command line.